src/bootstrap-table.js

```diff
@@ -2514,7 +2514,7 @@ class BootstrapTable {
           const column = this.columns[this.fieldsColumnsIndex[key]]
 
           if (!column) {
-            return
+            continue
           }
 
           row[key] = Utils.calculateObjectValue(column, this.header.formatters[column.fieldIndex], [value, row, row.index, column.field], value)
```

Comment:
Found a problem: since `row[key] = …` will change the original value, if getData is clicked multiple times, it will be executed multiple times, leading to incorrect results. 
The better way:
```js
      return data.map(row => {
        for (const [key, value] of Object.entries(row)) {
          const column = this.columns[this.fieldsColumnsIndex[key]]

          if (!column) {
            continue
          }

          return Utils.calculateObjectValue(column, this.header.formatters[column.fieldIndex],
            [value, row, row.index, column.field], value)
        }
      })
```

Reply:
Agreed, fixed.